src/libraries/System.IO.FileSystem/tests/FileStream/ctor_options.cs

```diff
@@ -162,21 +162,23 @@ public void PreallocationSize(FileAccess access, FileMode mode)
         [InlineData(FileMode.CreateNew)]
         public void WhenDiskIsFullTheErrorMessageContainsAllDetails(FileMode mode)
         {
-            const long tooMuch = 1024L * 1024L * 1024L * 1024L; // 1 TB
-
+            const long tooMuch = 1024L * 1024L * 1024L * 1024L * 1024L * 1024L; // 1 Exbibyte .. we are assuming this is not available
             string filePath = GetTestFilePath();
 
-            IOException ex = Assert.Throws<IOException>(() => CreateFileStream(filePath, mode, FileAccess.Write, FileShare.None, bufferSize: 1, FileOptions.None, tooMuch));
-            Assert.Contains(filePath, ex.Message);
-            Assert.Contains(tooMuch.ToString(), ex.Message);
-
-            // ensure it was NOT created
-            bool exists = File.Exists(filePath);
-            if (exists)
+            try
             {
-                File.Delete(filePath);
+                // not using Assert.Throws because in the event of failure, we want to dispose, so the next test isn't affected
+                using FileStream fs = CreateFileStream(filePath, mode, FileAccess.Write, FileShare.None, bufferSize: 1, FileOptions.None, tooMuch);
+                Assert.Fail($"Expected to throw IOException, {fs.Length}");
```

Comment:
Why not use Assert.Throws as @krwq suggested? e.g.
```C#
IOException ex = Assert.Throws<IOException>(() => CreateFileStream(filePath, mode, FileAccess.Write, FileShare.None, bufferSize: 1, FileOptions.None, tooMuch));
Assert.Contains(filePath, ex.Message);
Assert.Contains(tooMuch.ToString(), ex.Message);
Assert.False(File.Exists(filePath));
```
?

Reply:
If the test fails because the file is created, there is a race between the finalizer releasing the lock on it and the base class attempting to delete the file.

Now it's creating a colossal file, that's implausible unless there's some partial create mode in some future test configuration. But this way ensures the stream is disposed in that case.

In general we make reasonable attempts to not allow one test failure to trigger another, although of course that is not always feasible. That reduces noise in CI.

Reply:
That's my recollection of why I did this - what am I missing? But happy to change as you propose if you wish.

Reply:
IOW: The file created here will very likely occupy the whole disk in most configs. We need to dispose it, _even if this test fails_, to avoid causing out-of-disk errors to other tests.